addons/analytic/tests/test_analytic_account.py

```diff
@@ -211,3 +212,11 @@ def test_all_account_count_with_subplans(self):
             with self.subTest(plan=plan.name, expected_count=expected_value):
                 with Form(plan) as plan_form:
                     self.assertEqual(plan_form.record.all_account_count, expected_value)
+
+    def test_analytic_account_branches(self):
+        """
+        Test that an analytic account defined in a parent company is accessible in its branches (children)
+        """
+        self.analytic_account_1.company_id = self.company_data
+        self.analytic_account_1.with_company(self.company_b_branch).name = "AA changed by branch"
```

Comment:
I'm pretty sure this already worked before your PR because you have access to both companies in the environment anyway

Reply:
Running only the test in a no-demo db (no record rule changed)

```
2024-09-05 15:13:57,113 2166806 ERROR 3764627-17.0-no-demo odoo.addons.analytic.tests.test_analytic_account: ERROR: TestAnalyticAccount.test_analytic_account_branches
Traceback (most recent call last):
  File "/home/odoo/src/17.0/odoo/addons/analytic/tests/test_analytic_account.py", line 221, in test_analytic_account_branches
    self.analytic_account_1.with_company(self.company_b_branch).name = "AA changed by branch"
  File "/home/odoo/src/17.0/odoo/odoo/fields.py", line 1322, in __set__
    records.write({self.name: write_value})
  File "/home/odoo/src/17.0/odoo/addons/mail/models/mail_thread.py", line 317, in write
    result = super(MailThread, self).write(values)
  File "/home/odoo/src/17.0/odoo/odoo/models.py", line 4335, in write
    self.check_access_rule('write')
  File "/home/odoo/src/17.0/odoo/odoo/models.py", line 4139, in check_access_rule
    raise self.env['ir.rule']._make_access_error(operation, forbidden)
odoo.exceptions.AccessError: Uh-oh! Looks like you have stumbled upon some top-secret records.

Sorry, The anal(ytic) expert! (id=9) doesn't have 'write' access to:
- Analytic Account (account.analytic.account)
[...]
```